examples/us-dams/src/index.md

```diff
@@ -0,0 +1,320 @@
+---
+theme: ["cotton", "near-midnight"]
+---
+
+# Dam conditions by state
+
+```js
+const dams = FileAttachment("data/dam-simple.csv").csv({ typed: true });
+
+const fips = FileAttachment("data/county_fips_master.csv").csv({ typed: true });
+
+const capitals = FileAttachment("data/us-state-capitals.csv").csv({
+  typed: true,
+});
+
+const usCounties = FileAttachment("./data/us-counties-10m.json").json();
+```
+
+```js
+const fipsSelectedState = fips
+  .filter((d) => d.state_name == pickState)
+  .map((d) => d.fips);
+
+const capitalSelectedState = capitals.filter((d) => d.name == pickState);
+
+const states = topojson.feature(usCounties, usCounties.objects.states).features;
+
+const counties = topojson
+  .feature(usCounties, usCounties.objects.counties)
+  .features.map((d) => ({ ...d, fips: +d.id }));
+```
+
+```js
+const selectedState = states.filter((d) => d.properties.name === pickState);
+
+const selectedStateCounties = counties.filter((d) =>
+  fipsSelectedState.includes(d.fips)
+);
+```
+
+```js
+// Color palette for dam conditions
+const colors = ["#9498a0", "#ff725c", "#efb118", "#97bbf5", "#4269d0"];
+
+const conditions = [
+  "Not available",
+  "Poor",
+  "Unsatisfactory",
+  "Satisfactory",
+  "Fair",
+];
+```
+
+```js
+const pickState = view(
+  Inputs.select(
+    dams
+      .filter((d) => (d.state != "Guam") & (d.state != "Puerto Rico"))
+      .map((d) => d.state),
+    {
+      multiple: false,
+      label: "Pick a state:",
+      unique: true,
+      sort: true,
+      value: "Massachusetts",
+    }
+  )
+);
+```
+
+```js
+// Get dams just for the selected state
+const damsSelectedState = dams.filter((d) => d.state == pickState);
+```
+
+<div class="grid grid-cols-3 grid-rows-3" style="grid-auto-rows: 200px">
+<div class="card grid-colspan-2 grid-rowspan-3">
+<h2>${pickState} dams</h2>
+<h3>Map dot size represents maximum storage capacity (acre-feet). The scale should only be used to compare dam sizes within ${pickState}, not across states.</h3>
+${resize((width) => stackedBarChart(width))}
+${resize((width, height) => stateMap(width, height))}</div>
+<div class="card grid-rowspan-1" style="font-size: 1.2em">
+Of ${d3.format(",")(damsSelectedState.length)} NID recorded dams in ${pickState}, <b>${d3.format(",")(damsSelectedState.filter(d => d.conditionAssessment == "Poor").length)} ${damsSelectedState.filter(d => d.conditionAssessment == "Poor").length == 1 ? "is" : "are"} in poor condition</b>. Of those in poor condition, <b>${d3.format(",")(damsSelectedState.filter(d => d.conditionAssessment == "Poor" && d.hazardPotential == "High").length)} ${damsSelectedState.filter(d => d.conditionAssessment == "Poor" && d.hazardPotential == "High").length == 1 ? "has" : "have"} high hazard potential</b>, where "downstream flooding would likely result in loss of human life." 
```

Comment:
```suggestion
Of ${d3.format(",")(damsSelectedState.length)} NID recorded dams in ${pickState}, <b>${d3.format(",")(damsSelectedState.filter(d => d.conditionAssessment == "Poor").length)} ${damsSelectedState.filter(d => d.conditionAssessment == "Poor").length == 1 ? "is" : "are"} in poor condition</b>. Of those in poor condition, <b>${d3.format(",")(damsSelectedState.filter(d => d.conditionAssessment == "Poor" && d.hazardPotential == "High").length)} ${damsSelectedState.filter(d => d.conditionAssessment == "Poor" && d.hazardPotential == "High").length == 1 ? "has" : "have"} high hazard potential</b>, where “downstream flooding would likely result in loss of human life.”
```

Also, this block of text tends to overflow out of its card—which given the subject is worrying :)

Reply:
Yeah I didn't figure out a good solution for this yet, still digging. Another issue is that when it's resized (e.g. for mobile) the text takes up just a small fraction of the container, which looks bad